### PR TITLE
[doc][VM] Update the vm doc

### PR DESCRIPTION
**Pull request by zhiics**
https://discuss.tvm.ai/t/relay-vm-from-c/5623

The VM doc is stale, this PR updates it.

@icemelon9 we need a followup to update the dynamic shape handling section.

@wweic @jroesch @jmorrill

### CODE REVIEW

File: docs/dev/virtual_machine.rst [hunk 1]
```diff
@@ -94,89 +94,115 @@ Ret
...
+AllocStorage
+^^^^^^^^^^^^
+**Arguments**:
+::
+  RegName dst
+  RegName size
+  RegName alignment
+  DLDataType dtype_hint
+
+Allocate a storage block with the given ``size``, ``alignment`` and and data type, ``dtype_hint``.
```
Comment (wweic) [inline docs/dev/virtual_machine.rst]:
```suggestion
Allocate a storage block with the given ``size``, ``alignment`` and data type, ``dtype_hint``.
```

File: docs/dev/virtual_machine.rst [hunk 2]
```diff
@@ -284,40 +297,76 @@ Dispatch Loop
...
+This loop is implemented by ``VirtualMachine::Run()``.
 
 VM Compiler
 ~~~~~~~~~~~
 
 An important part of this infrastructure is a compiler from Relay's full IR into a sequence of bytecode.
-The VM compiler transforms a `tvm::relay::Module` into a `tvm::relay::vm::VirtualMachine`. The virtual
-machine contains a set of compiled functions, the compiled functions are contained in `tvm::relay::vm::Function`. The functions contain metadata about the the function as well as its compiled bytecode. For full definitions of the data structures see `vm.h`.
+The VM compiler transforms a ``tvm::relay::Module`` into a ``tvm::relay::vm::Executable``. The executable
+contains a set of compiled functions, the compiled functions are contained in ``tvm::relay::vm::Function``. The functions contain metadata about the the function as well as its compiled bytecode. The emitted executable object then can be loaded and run by a ``tvm::relay::vm::VirtualMachine`` object. For full definitions of the data structures, please see `include/tvm/runtime/vm.h`_.
```
Comment (wweic) [inline docs/dev/virtual_machine.rst]:
```suggestion
contains a set of compiled functions, the compiled functions are contained in ``tvm::relay::vm::Function``. The functions contain metadata about the function as well as its compiled bytecode. The emitted executable object then can be loaded and run by a ``tvm::relay::vm::VirtualMachine`` object. For full definitions of the data structures, please see `include/tvm/runtime/vm.h`_.
```

File: docs/dev/virtual_machine.rst [hunk 3]
```diff
@@ -284,40 +297,76 @@ Dispatch Loop
...
 - A-Normal Form
-- Lambda Lift (see `src/relay/vm/lambda_lift.cc`)
-- Inline Primitives (see `src/relay/vm/inline_primitives.cc`)
-- Inliner (see `src/relay/pass/inliner.cc`)
-- Constant Pool Layout (see `src/relay/backend/vm/compiler.cc`)
-- ADT Tag Allocation (see `src/relay/backend/vm/compiler.cc`)
+- Lambda Lift (see `src/relay/vm/lambda_lift.cc`_)
+- Inline Primitives (see `src/relay/vm/inline_primitives.cc`_)
+- Constant Pool Layout (see `src/relay/backend/vm/compiler.cc`_)
+- ADT Tag Allocation (see `src/relay/backend/vm/compiler.cc`_)
```
Comment (wweic) [inline docs/dev/virtual_machine.rst]: What is "ADT Tag Allocation"?

Comment (zhiics) [reply docs/dev/virtual_machine.rst]: Yeah, it was from the old version. We need to remove it.

**Comment by wweic** [comment]
thanks @zhiics @tqchen

**Pull request merged**